# Active and Passive Management - I

## Introduction
This module introduces you to the some of the methods used to measure the performance of portfolio managers. Since the vast majority of capital in financial markets is managed by professional investors, being able to measure the quality a portfolio manager is essential to maintaining an efficient market. For instance, if investors were unable to assess a portfolio managers performance, investors could continue to invest with the and this would result in an inefficient allocation of capital. The competition for investors funds via well performing portfolios is what keeps the allocation of capital efficient. Hence the evaluation of a portfolio, and by extension its manager, is a crucial task in finance.

The difficulty with determining the performance of a portfolio is understanding how to trade off the return that a manager generates against the risk that the manager takes to obtain their return. For example, a portfolio anager could take an extremely risky position and make a huge amount of money from it. Looking at retruns earned, they would appear to be extremely talented. However, they could also have been lucky that the risk they took on didn't materialise. A good portfolio manager is one which is able to consistently earn good returns without having to take on a a disproportionate amount of risk. Unfortunately simply stating that we have to consider risk isn't sufficient. What kind of risk? Total risk, systematic risk or perhaps another kind? As we will see, the measure of performance is somewhat driven by the investment approach of the portfolio manager.

Significant issues arise when the portfolio manager changes the composition of the portfolio. Usually managers change portfolios in an effort to "time" the market (increase investment when the market is going up and reduce it when its going down). Market timing is very difficult to do successfully but the ability to do it successfully is extremely valuable. We will examine several approaches to determine if a portfolio manager is able to successfully time the market. Finally, often times we are more concerned with what decisions led to a portfolios performance, whether it be good or bad. If we can identify what decisions were made well (led to good performance) and those that were made poorly (led to poor performance) then we can take corrective action. Performance attribution is a technique that can allow the investor to understand where portfolio performance comes from.

## Computing Returns - The Conventional Theory
Before we consider risk adjusted returns, we first consider how one can compute "average" returns that have been earned by a portfolio. Perhaps the most obvious approach is to compute the arithmetic average of returns, $\bar{r}$. This is given by
\begin{align}
    \bar{r} = \frac{1}{T}\sum_{t=1}^{T}r_{t}
\end{align}
The arithematic average is a good measure of the central tendency (expected vaule) of a distribution and hence is often used as an input into portfolio optimisation models that require an estiamte of the future expected value. However, the downside of the arithmetic average is that it is not necessariliy a return that can be realised. To see this, consider for example a portfolio that starts at \\$100 at $t=0$, goes up to \\$150 at $t=1$ and back down to \\$100 at $t=2$. The returns per period are
\begin{align}
    r_{1} &= \frac{150 - 100}{100} = 0.5\\
    r_{2} &= \frac{100-150}{150} = -0.33
\end{align}
The arithmetic average of these two returns is $\bar{r} = \tfrac{1}{2}(0.5 -0.33) = 0.085$ and it appers that we have profited when in reality we have earned no money whatsoever. The reason for this discrepancy is that the reative gain in period 1 was larger than the relative loss in period 2.

The geometric average can overcome this problem as it corresponds to a return which can be earned. The geometric average, $\bar{r}_{g}$ is given by the expression
\begin{align}
    (1+\bar{r}_{g})^{T} = \prod_{t=1}^{T}(1+r_{t}) \quad \Rightarrow \quad \bar{r}_{g} = \left(\prod_{t=1}^{T}(1+r_{t})\right)^{1/T} - 1
\end{align}
Inspecting the above equation, it is clear that the geometric average corresponds to the $T$-period compounded return and hence earning the fixed return $\bar{r}_{g}$ for $T$ periods would yield the same wealth as the sequence of returns $\{r_{1}, r_{2}, \ldots, r_{T}\}$ where wealth is reinvested. One thing to note is that we do not weight one time period more or less than any other when computing the geometric average and for this reason it is sometimes called a "time-weighted" return. However, there may be times where we do wish to weight one time period more than another. This leads to dollar-weighted returns.

Dollar weighting returns simply applies a greater weighting to a periods return if we have a greater amount of wealth invested as that periods return is more important in determining ur overall performance. While at first glance appearing ike a complicated task, the tool required to compute dollar-weighted returns is already familiar, the internal rate of return. The key to computing dollar-weighted returns is to treat the investment like a project with capital flows in and out. Consider for example the problem where cash is withdrawn or added to a portfolio. Imagine that you observe the following decisions made at time $t=\{0, 1, 2\}$.


<table>
    <tr>
    <th style="text-align:left">$t$</th>
    <th style="text-align:left">Outlays</th>
    </tr>
    <tr>
    <td style="text-align:left">0</td>
    <td style="text-align:left">\$50 to purchase first share </td>
    </tr>
    <tr>
    <td style="text-align:left">1</td>
    <td style="text-align:left">\$53 to purchase second share a year later</td>
    </tr>
    <tr>
    <th style="text-align:left"></td>
    <th style="text-align:left">Proceeds</td>
    </tr>
    <tr>
    <td style="text-align:left">1</td>
    <td style="text-align:left">\$2 dividend from initially purchased share</td>
    </tr>
    <tr>
    <td style="text-align:left">2</td>
    <td style="text-align:left">\$4 dividend from the two shares held in time 2</td>
    </tr>
    <tr>
    <td style="text-align:left">2</td>
    <td style="text-align:left">\$108 received from selling the two shares at time 2</td>
    </tr>
</table>

Using the discounted cash flow (DCF) approach, we may compute the average return earned by equating the present value of outlays and proceeds,
\begin{align}
    50 + \frac{53}{1+r} = \frac{2}{1+r} + \frac{112}{(1+r)^{2}} \quad \Rightarrow \quad -50 - \frac{51}{(1+r)} + \frac{112}{(1+r)^{2}} = 0
\end{align}
Solving for the discount rate where these present values are the same gives the internal rate of return (IRR) which is the dollar weighted return $r_{d}$. The solution to the above problem is $r_{d}=7.117\%$. While this simple example could be solved analytically, for more complex cashflows a numerical solution is required. Note that the dollar-weighted return $r_{d}$ is different from the time-weighted return $r_{g}=7.81$\%. This is because the stocks performance in the second period, when more money is invested, is worse than the first period, dragging the dollar weighted return down.

### Python Example: IRR
The code below computes the IRR for the sequence of cashflows in the example provided.

In [1]:
# Import the numpy financial package.
# This may need to be downloaded into your python installation.
# If so, open the python command prompt and type "pip install numpy_financial" then press enter.
import numpy_financial as npf

# Compute the IRR for the above problem and round to 5 decimal places.
round(npf.irr([-50, -51, 112]),5)

0.07117

## Adjusting for Risk
On their own, the measures of average return (arithmetic average, time-weighted and dollar-weighted) are insufficient to assess a portfolio managers performance as they don't account for risk. One simple approch to making a risk adjustment is to simply rank managers who have similarly risky portfolios. One could then argue that the median manager is an appropriate benchmark as half of managers outperform this and the other half underperform. However, such a simple analysis can miss some of the nuances associated with risk. Take equity investment for example. While it may seem like all equity investments would carry a similar risk, different managers can focus on different subsets of equity (small cap, high beta, growth stocks, etc) and hence risk can vary across this universe. A better approach is to compute measure of risk adjusted performance.

## Risk Adjusted Performance Measures
We begin by outlining the four most common performance measures:

1. The Sharpe ratio: $\frac{\bar{r}_{P} - \bar{r}_{f}}{\sigma_{P}}$
2. The Treynor ratio: $\frac{\bar{r}_{P} - \bar{r}_{f}}{\beta_{P}}$
3. Jensens alpha: $\alpha_{P} = \bar{r}_{P} - (\bar{r}_{f} + \beta(\bar{r}_{M} - \bar{r}_{f}))$
4. The information ratio: $\frac{\alpha_{P}}{\sigma_{e_{P}}}$

Note that each performance measure uses either a different measure for performance ($\bar{r}_{P} - \bar{r}_{f}$, $\alpha_{P}$) and/or risk ($\sigma_{P}$, $\beta_{P}$, $\sigma_{e_{P}}$). Each measure has its appeal, but similarly, each measure has its limitations. The user of these measures should be aware of which circumstances allow the appropriate use of each of these measures.

| Performance Measure | Definition | Application |
|:--|:-:|:--|
| Sharpe | $\frac{\bar{r}_{P} - \bar{r}_{f}}{\sigma_{P}}$ | Choosing among portfolios competing for the overall risky portfolio |
| Treynor | $\frac{\bar{r}_{P} - \bar{r}_{f}}{\beta_{P}}$ | Whan ranking portfolios that will be mixed to obatain the overall risky portfolio |
| Information Ratio | $\frac{\alpha_{P}}{\sigma_{e_{P}}}$ | When evaluating a portfolio to be mixed with the benchmark portfolio |

## Interpreting Performance Measures
While the Sharpe and Treynor ratios can be easily used to rank competing portfolios, we are also interested in understanding what the differences between two ratios means economically. For example, say we have two portfolios $P$ and $M$ with respective Sharpe ratios $S_{P} = 0.69$ and $S_{M} = 0.73$. While we can say that $P$ underperformed $M$ (as measured by the Sharpe ratio) we don't know what the difference of 0.04 means economically. To address this, Leah Modigliana and her grandfather Franco developed the $M^{2}$ measure. The idea is to combine an active portfolio ($P$ say) with a risk-free asset (T-bills) to create a new portfolio $P^{*}$. We select the the amount of the risk-free asset such that the standard deviation (risk) of $P^{*}$ is equal to the benchmark portfolio $M$. Because these portoflios now have the same risk, we can compare their performance by comparing their returns. Hence, the $M^{2}$ measure is given by
\begin{align}
    M^{2} = r_{P^{*}} - r_{M}
\end{align}

We can do the exact same exercise but use systematic risk ($\beta$) in place of total risk ($\sigma$) and compare the returns of portfolios with the same $\beta$. This is somtimes called the $T^{2}$ measure.

In [2]:
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import figure
from ipywidgets import interact

r_f = 0.06
r_p = 0.35
r_m = 0.28
b_p = 1.2
b_m = 1
s_p = 0.42
s_m = 0.3
se_p = 0.18
se_m = 0

# Find the weight in rf needed to make the standard deviations equal
x_rfstar = 1 - s_m/s_p
M2 = x_rfstar*r_f + (1 - x_rfstar)*r_p - r_m
print("The weight allocated to the risk free asset is ", round(x_rfstar,5))
print("The M^2 measure is " + str(round(M2, 5)))

@interact
def CL(w_rf=(0,1,0.05)): # The weight allocated to the ris
    
    
    # Compute the levered portfolio pstar
    r_pstar = w_rf*r_f + (1 - w_rf)*r_p
    s_pstar = (1 - w_rf)*s_p
    b_pstar = (1 - w_rf)*b_p
    
    # Compute CML for plotting
    x_rf = np.linspace(-0.5, 1.5, 100)
    CML_r = x_rf*r_f + (1-x_rf)*r_m
    CML_s = (1-x_rf)*s_m
    
    # Compute SML for plotting
    CML_r = x_rf*r_f + (1-x_rf)*r_m
    CML_b = (1-x_rf)*b_m
    
    # Create the M^2 plot
    plt.plot(CML_s, CML_r, 'k--', label = "CML")
    plt.plot(s_m, r_m, 'ro', label = "m")
    plt.plot(s_p, r_p, 'bo', label = "p")
    plt.plot(s_pstar, r_pstar, 'gs', label = "p*")
    plt.xlabel('$\sigma$')
    plt.ylabel('$r$')
    plt.legend(loc="upper left")
    plt.grid(True)
    plt.xlim([0, max(s_p,s_m)+0.05])
    plt.ylim([0, max(r_p,r_m)+0.05])

    return plt.show()
   

The weight allocated to the risk free asset is  0.28571
The M^2 measure is -0.01286


interactive(children=(FloatSlider(value=0.0, description='w_rf', max=1.0, step=0.05), Output()), _dom_classes=…

## Alpha and Performance Measures
We examine now how alpha impacts performance measures. Recall that alpha ($\alpha$) represents the extra expected return earned over and above what you should expect to earn for bearing systematic risk. A portfolio obtains positive alpha by overweighting undervalued securities. You may wonder then why alpha doesn't play a bigger role in determining the ability of a fund manager. In reality, it does. Assuming returns are described by the CAPM/single factor model,
\begin{align}
    r_{p} - r_{f}= \alpha + \beta(r_{m} - r_{f}) + e
\end{align}
We can show that alpha is related to the Sharpe and Treynor ratios in the following way

|   | &nbsp; &nbsp; &nbsp; Treynor ($T_{P}$) &nbsp; &nbsp; &nbsp;| &nbsp; &nbsp; &nbsp; &nbsp; Sharpe ($S_{P})$ &nbsp; &nbsp; &nbsp; &nbsp;| Information |
|:--|:-:|:-:|:-:|
| Relation to $\alpha$ | $T_{P} = \frac{\alpha}{\beta} + T_{M}$ | $S_{P} = \frac{\alpha}{\beta} + \rho S_{M}$ | $\frac{\alpha}{\sigma_{e_{P}}}$ |
| Improvement over market benchmark | $\frac{\alpha}{\beta}$  | $\frac{\alpha}{\beta} - (1 - \rho)S_{M}$  | $\frac{\alpha}{\sigma_{e_{P}}}$ |

What is clear from the table above is that, holding all other variables constant, any improvement in alpha will improve the measured performance of the portfolio. It is important to note that while alpha is necessary for good performance, it is not sufficient to guarantee good performance. If alpha comes at the cost of increased risk because of limited diversification, the portfolio performance may suffer.

### Example: Performance Measures in Practice
The python code below computes the performance measures discussed for two funds: the Fidelity Magellan Fund and the Vanguard US Growth Fund. First, lets look at the data.

In [3]:
# Import the pandas package and refer to is as "pd"
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm

# Load the .csv file which has been saved locally.
# Column separators are denoted with "," and the headers for each column are on line 0.

# Data = pd.read_csv("rC:\Users\hmal4958\Dropbox\Teaching\FINC6001\Fund_Data_Topic_9.csv", sep = ",", header = 0)    
Data = pd.read_csv("E:\Dropbox\Teaching\FINC6001\Fund_Data_Topic_9.csv", sep = ",", header = 0)
Data



Date  Fidelity Magellan  Vanguard U.S. Growth Fund   rM-rf      rf
0   20210201           0.009486                   0.014790  0.0279  0.0000
1   20210101          -0.030651                   0.003710 -0.0004  0.0000
2   20201201           0.029992                   0.008844  0.0463  0.0001
3   20201101           0.081058                   0.134316  0.1247  0.0001
4   20201001          -0.034596                  -0.035833 -0.0210  0.0001
5   20200901          -0.036508                  -0.043821 -0.0363  0.0001
6   20200801           0.074169                   0.112869  0.0763  0.0001
7   20200701           0.074176                   0.087760  0.0577  0.0001
8   20200601           0.025352                   0.063699  0.0246  0.0001
9   20200501           0.070352                   0.089706  0.0558  0.0001
10  20200401           0.129398                   0.170916  0.1365  0.0000
11  20200301          -0.097336                  -0.105537 -0.1338  0.0012
12  20200201          -0.069590                  -0.057676 -0.0813  0.0012
13  20200101           0.031430                   0.047981 -0.0011  0.0013
14  20191201          -0.103691                  -0.015900  0.0277  0.0014
15  20191101           0.036430                   0.051258  0.0387  0.0012
16  20191001           0.011981                   0.025424  0.0206  0.0015
17  20190901          -0.007319                  -0.012041  0.0143  0.0018
18  20190801           0.005520                  -0.016742 -0.0258  0.0016
19  20190701           0.018744                   0.012645  0.0119  0.0019
20  20190601           0.061209                   0.073720  0.0693  0.0018
21  20190501          -0.057998                  -0.059495 -0.0694  0.0021
22  20190401           0.044966                   0.047812  0.0396  0.0021
23  20190301           0.020958                   0.016707  0.0110  0.0019
24  20190201           0.028748                   0.044246  0.0340  0.0018
25  20190101           0.091089                   0.099974  0.0841  0.0021
26  20181201          -0.135005                  -0.170927 -0.0955  0.0019
27  20181101           0.015671                   0.026836  0.0169  0.0018
28  20181001          -0.097259                  -0.092525 -0.0768  0.0019
29  20180901           0.005333                   0.002048  0.0006  0.0015
30  20180801           0.034293                   0.048758  0.0344  0.0016
31  20180701           0.033542                   0.021857  0.0319  0.0016
32  20180601           0.005100                   0.012564  0.0048  0.0014
33  20180501          -0.019729                   0.044835  0.0265  0.0014
34  20180401           0.010678                   0.003140  0.0029  0.0014
35  20180301          -0.027300                  -0.019273 -0.0235  0.0012
36  20180201          -0.035721                  -0.018334 -0.0365  0.0011
37  20180101           0.085475                   0.088008  0.0558  0.0011
38  20171201          -0.032118                  -0.042219  0.0106  0.0009
39  20171101           0.023106                   0.023841  0.0312  0.0008
40  20171001           0.033976                   0.030552  0.0225  0.0009
41  20170901           0.023962                   0.014853  0.0251  0.0009
42  20170801           0.011870                   0.019339  0.0016  0.0009
43  20170701           0.029667                   0.035236  0.0187  0.0007
44  20170601           0.008840                   0.003099  0.0078  0.0006
45  20170501          -0.035026                   0.029780  0.0106  0.0006
46  20170401           0.013166                   0.030443  0.0109  0.0005
47  20170301          -0.000206                   0.014328  0.0017  0.0003
48  20170201           0.039333                   0.036222  0.0357  0.0004
49  20170101           0.028978                   0.042463  0.0194  0.0004
50  20161201          -0.009540                  -0.015800  0.0182  0.0003
51  20161101           0.029694                  -0.005067  0.0486  0.0001
52  20161001          -0.019376  

It should be clear that we have about 5 years worth of monthly returns on each fund, the excess return on the market portfolio and the risk free rate. Now, lets convert the data into a numpy array and do some regressions. Note that while we could output all the regresion results, we are only going to save what we need, alpha and beta, and store them as variables to be used to compute the performance measures.

In [4]:
npData = Data.to_numpy() # Convert data frame to numpy array
Y1 = npData[:,1]-npData[:,4] # This is Fidelity excess returns
Y2 = npData[:,2]-npData[:,4] # This is Vanguard excess returns
X = npData[:,3] # This is the markets excess return
X = sm.add_constant(X) # Add in the constant term to the regression.
RegModel1 = sm.OLS(Y1, X) # Create the regression model.
RegModel2 = sm.OLS(Y2, X) # Create the regression model.

# Obtain results by fitting the data.
RegResults1 = RegModel1.fit()
RegResults2 = RegModel2.fit()

# Store tha alpha and beta coefficients
alpha1 = RegResults1.params[0]
beta1 = RegResults1.params[1]

alpha2 = RegResults2.params[0]
beta2 = RegResults2.params[1]

sigma_e1 = np.sqrt(np.var(Y1 + npData[:,4]) - beta1**2*np.var(X[:,1]))
sigma_e2 = np.sqrt(np.var(Y2 + npData[:,4]) - beta2**2*np.var(X[:,1]))

# Compute the performance measures and store as strings
SharpeF = str(round(np.mean(Y1)/np.var(Y1 + npData[:,4]),5))
SharpeV = str(round(np.mean(Y2)/np.var(Y2 + npData[:,4]),5))

TreynorF = str(round(np.mean(Y1)/beta1,5))
TreynorV = str(round(np.mean(Y2)/beta2,5))

JensenAlphaF = str(round(alpha1,5))
JensenAlphaV = str(round(alpha2,5))

InfRatioF = str(round(alpha1/sigma_e1,5))
InfRatioV = str(round(alpha2/sigma_e2,5))

# Print out the results
print('The Sharpe ratios are:')
print('Fidelity: '+ SharpeF + ' and Vanguard: ' + SharpeV)
print()
print('The Treynor ratios are:')
print('Fidelity: '+ TreynorF + ' and Vanguard: ' + TreynorV)
print()
print('The Jensen alpha are:')
print('Fidelity: '+ JensenAlphaF + ' and Vanguard: ' + JensenAlphaV)
print()
print('The Information ratios are:')
print('Fidelity: '+ InfRatioF + ' and Vanguard: ' + InfRatioV)



The Sharpe ratios are:
Fidelity: 2.91472 and Vanguard: 4.90717

The Treynor ratios are:
Fidelity: 0.00764 and Vanguard: 0.014

The Jensen alpha are:
Fidelity: -0.00533 and Vanguard: 0.00069

The Information ratios are:
Fidelity: -0.21947 and Vanguard: 0.02967


We can see from these results that the Vanguard portfolio appears to be a superior portfolio to the Fidelity portfolio by every metric we have examined.

## Style Ananlysis
Style analysis was introduced by William Sharpe as a tool to measure the systematic exposure that a fund has to different classes of assets. In much the same way that we can  regress the returns of a portfiolio/asset against risk fatcors to identify its exposures, we can also regress a funds returns against other more basic asset classes to infer the style of investment that the fund employs. For example, a fund that advertises itself as tracking the S\&P 500 should have a beta of 1 with respect to te  S\&P 500 returns and a beta of 0 with respect to other asset classes (bonds, small cap equity, T-bills, etc). By performing these regressions we can understand what kinds of asset classes, and hence what style of investment, the fund employs. Unfortunately, we cannot use a standard regression as non-zero correlations can interfere with the estimates. However, by constrainng the estimatation to ensure that all beta coefficients add up to 1 and are greater than 0 (for long only funds) we can easily circumvent this problem.

The results from Sharpe's initial analysis was striking. It found that over 90\% of the variation ($R^{2}$) in a funds returns could be explained by the borad asset classes the fund invests in. More recent research continues to validate numbers like this. This result sent shockwaves through the funds management industry as it implied that in order to replicate the performance of a professional manager, all you needed to do was select the appropriate mix of broad asset classes. The specific assets within those classes appeared to be unimportant to fund performance. We call the selection of broad asset classes the *asset allocation decision* while the selection of specific securities within each asset class is the *security selection decision*. The work of Sharpe seemed to indicate that only asset allocation really matters.

The interpretation that fund managers provide no benefit beyond the broad asset classes isn't quite correct though as it ignores the intercepts of the regression. If the intercept is positive and significant, this would suggest that despite carrying the risk associated with the broad asset classes to which it is exposed, the manager has been able to extract extra return over and above what might be expected given the risk of the asset classes. The presence of alpha would then suggest that a manager has skill at security selection. What is not debated though is that asset allocation makes up a huge part of a funds performance.

### Example: Style analysis of Fidelity Magellan Fund
Using data for the Fidelity Magellan fund, the example below performs a constrained regression to identify its style. We only use size and value portfolios (all from the Ken French Data Library) as our asset classes.

In [9]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from scipy.optimize import minimize
from scipy.optimize import Bounds
from scipy.optimize import LinearConstraint

# Load the data and organize into numpy arrays
Data = pd.read_csv("E:\Dropbox\Teaching\FINC6001\Style_Analysis_Data.csv", sep = ",", header = 0)
npData = Data.to_numpy()
Y = npData[:,1]
X = npData[:,2:9]
X = sm.add_constant(X)

# Define coefficients for quadratic objective function
A=X.T@X
b=-2*Y@X
c=Y@Y

# Define the objective function
def obfun(x,A,b,c):
    return x@A@x + b@x + c

# Specify the constraints
n = A.shape[1]
lb = np.zeros(n)
ub = np.ones(n)
bnds = Bounds(lb,ub)
linear_constraint = LinearConstraint(np.ones(n), 1, 1)
x0 = 1/n*np.ones(n)

# Optimize
res = minimize(obfun,x0, args=(A, b, c), bounds=bnds, constraints=[linear_constraint])

# Compute the R-square
weights = res.x[1:]
Rsqr = round(np.var(weights@X[:,1:].T)/np.var(Y),3)

# Print the results
DataHeaders = Data.columns.values
DataHeaders = DataHeaders[2:]
DataHeaders = np.insert(DataHeaders,0,"alpha")
Coefficients = np.round(res.x,3)
Results = np.vstack((DataHeaders,Coefficients))
print(Results.T)
print()
print("The R-squared for this constrained regression is " + str(Rsqr))

[['alpha' 0.0]
 ['Small Cap' 0.0]
 ['Med Cap' 0.0]
 ['Large Cap' 0.534]
 ['Low PE' 0.4]
 ['Med PE' 0.0]
 ['High PE' 0.0]
 ['T-Bills' 0.065]]

The R-squared for this constrained regression is 0.702


As we can see from the output above, the Fidelity Magellan fund invests in low PE/large cap firms with a small amount invested in T-bills. The $R^{2}$ from this regression is about 70\% suggesting that you can obtain returns on a portfolio that explains 70\% of the Fidelity Magellan funds variance by investing in the 7 provided asset classes. This number is relatively low and hence suggests that there may be asset classes that we haven't accounted for.

## Changing Composition & Market Timing
All the performance metrics considered have a common flaw. They all assume that the risk is constant over the period of the measurement. This isn't necessarily how manager behave though. If a manager believes that the equity market is about to go up, they may shift their portfolio towards equities and hence reduce their investment in other assets. This will cause the risk of their portfolio to change through time. This can lead to difficulties using the oreviously outlined performance measures.

We can however test to see if a manager is successfully altering their portfolio allocations for the benefit of fund performanc, or if they are making alterations for another, less noble reason (see Section 24.3 of the text). This shift of wealth from an underperforming asset into an outperforming onw is called *market timing*. In its purest form, market timing involves moving moey into the equity market when it is predicted to outperform and back into the risk-free asset when the market is expected to underperform.

To consider how we might test if a fund maager possess market timing ability, consider how the $\beta$ of a portfolio changes as the proportion of wealth invested in the equity market changes. The portfolio beta, $\beta_{P}$, is simply the weighted sum of all constituent betas. Say a portfolio is composed of the market portfolio and the risk free asset, then the portfolio beta is given by
\begin{align}
    \beta_{P} = w_{r_{f}}\beta_{r_{f}} + (1 - w_{r_{f}})\beta_{M}
\end{align}
Since $\beta_{r_{f}} = 0$ and $\beta_{M} = 1$, we have that
\begin{align}
    \beta_{P} = (1 - w_{r_{f}})
\end{align}
and hence as the allocation to the risk free asset goes down (or equivalently as the allocation to the market goes up) the portfolio beta increases. We may visualise $\beta_{P}$ as the slope of the line of best fit obtained when plotting the returns of the asset against the returns of the market (the characteristic line). Hence, if this slope changes, it could be a sign of market timing.

There are several approaches used to test for this change in slope. These involve running the regressions
\begin{align}
    r_{P,t} - r_{f,t} &= a + b(r_{M,t} - r_{f,t}) + c(r_{M,t} - r_{f,t})^{2} + e_{t}\\
    r_{P,t} - r_{f,t} &= a + b(r_{M,t} - r_{f,t}) + c(r_{M,t} - r_{f,t})D(r_{M,t} - r_{f,t} > 0) + e_{t}\\
\end{align}
where $D(x>0)$ is a dummy variable that is 1 if $x>0$ and 0 otherwise. Both of these approaches test that if the market return is large, does the beta of the portfolio increase? If the coefficient $c$ is positive and statisticaly significant, this would suggest that when the market is ourperforming the portfolio beta is increased relative to when it is underperforming. If the coeficient $c$ isn't statistically different from 0 however, this would suggest no eveidence of market timing.

It is natural to ask if it is worth that much to time the market. To understand the potential value of market timing, we can study the returns of a paper portfolio that invests in the market if its return was greater than return on T-bills and invests in T-bills if their return is larger. This procedure naturally assumes that the investor has perfect timing ability. Investments are made annually at the start of each year and begin in 1927 starting with \$1. The results from this exercise are tabulated below:

|   | T-Bills | Equity Market | Perfect Timer |
|:--|:-:|:-:|:-:|
| Terminal Value | \$20  | \$5,721  | \$755,809  |
| Arithemetic Average | 3.39\%  | 11.49\% | 16.41\% |
| Geometric Average  | 3.34\%  | 9.76\% | 15.85\% |
| Standard Deviation  | 3.14\% | 20.04\% | 13.44\% |

Clearly the ability to time the market perfectly is an extremely vauable skill to posess. Another way of thinking about maket timing skill is through a derivative called a call option. Call options are derivative securities that give the holder the right, but not the obligation, to purchase a security at an agreed upon price (strike) at a time in the future (maturity). Naturally, if the price of the security is greater than the strike at maturity, the option holder will exercise the option, purchase for the strike price and immediatley sell in the market for the higher market price, netting them a profit. If the market price at maturity is below the strike, the investor will simply elect to not exercise their option. In this case the payoff from having this market timing option is
\begin{align}
    \max(S_{T}, S_{0}(1+r_{f}))
\end{align}
where $S_{T}$ is the value of the market at time $T$ and $S_{0}$ is the value of the portfolio at inititation. Options don't come free though, you have to pay a premium for them. The value of the ability to time the market then is equivalent to the present value of this call option. Using the Black-Scholes model (for which he was partly responsible) Robert Merton showed that the present value of perfect market timing ability can be written as
\begin{align}
    PV(\textrm{Perfect Timer}) = 2N\left(\tfrac{1}{2}\sigma_{M}\sqrt{T}\right)
\end{align}
where $N(\cdot)$ is the normal cumulative density function, $\sigma_{M}$ the market volatility and $T$ the time to maturity.

The ability to time the market perfectly is essentially impossible to posess. Hence we ask the question, what is the value of being an imperfect timer? By this we mean what is the value of the ability to time the market is the prediction is correct with some probability $P$. However, there are two states to predict, these being a bull market (market outperforms) and a bear market (market underperforms). Let $P_{1}$ be the correct proportion on bull predictions and $P_{2}$ be the correct proportion of bear predictions. Then Merton showed that the value of market timing is may be written as
\begin{align}
    PV(\textrm{Imperfect Timer}) = (P_{1} + P_{2} - 1)2N\left(\tfrac{1}{2}\sigma_{M}\sqrt{T}\right)
\end{align}
Since $(P_{1} + P_{2} - 1) < 1$ for any form of imperfect timimg, this term reduces the value of the call option which corresponds to the value of market timing skill. As an example, say an investor could predict bull and bear markets with 70\% accuracy. Then their value of market timing would be only 40\% as valuable as someone who can time perfectly. AN investor who is basically flipping a coin ($P_{1} = P_{2} = 0.5$) adds no value through market timing.

### Python example: Market timing estimation
The code below performs the market timing regressions using the data previously used for style analysis.

In [63]:
Data = pd.read_csv("E:\Dropbox\Teaching\FINC6001\Fund_Data_Topic_9.csv", sep = ",", header = 0)
npData = Data.to_numpy()
rf = npData[:,4]
Fidelity = npData[:,1] - rf
Vanguard = npData[:,2] - rf
Mkt = npData[:,3]
MktSqr = Mkt**2
MktD = Mkt*(Mkt > 0)
X1 = np.vstack((Mkt, MktSqr)).T # Stack the arrays on top op each other the transpose to get the right size array
X1 = sm.add_constant(X1)

X2 = np.vstack((Mkt, MktD)).T
X2 = sm.add_constant(X2)

# Change the inputs to run the desired regression
RegModel = sm.OLS(Fidelity, X1) # Create the regression model.
RegResults = RegModel.fit()
print(RegResults.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.753
Model:                            OLS   Adj. R-squared:                  0.744
Method:                 Least Squares   F-statistic:                     82.36
Date:                Fri, 16 Apr 2021   Prob (F-statistic):           3.96e-17
Time:                        09:19:30   Log-Likelihood:                 130.32
No. Observations:                  57   AIC:                            -254.6
Df Residuals:                      54   BIC:                            -248.5
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.0058      0.004     -1.477      0.1

We can see from the provided regression output that Fidelity doesn't appear to have market timing ability as the p-value attached to the coefficient of x2 is greater than 0.05. Try adjusting the code to get the results for other market timing models and the Magellan fund.

## Performance Attribution
Rather than simply focusing on total portfolio performance, oftentimes we are interested in which decisions led to the portfolio performance. Portfolios are constructed by making decisions about the borad asset classes in which to invest (the asset allocation decision), the sectors witihn those classes and then the securities within those sectors (the security selection decision). We can of course define more granular decisions but these three are the most commonly identified. The aim of performance attribution is to explain the difference between a benchmark portfolio $B$ (sometimes called a bogey portfolio) and the portfolio managers portfolio $P$ as a consequence of these decisions.

Say for example that the benchmark equity portfolio is the S\&P 500 index, the benchmark fixed income portfolio is the Barcalay Aggregate index and the benchmark cash rate is the rate earned in the money market (LIBOR). Each one of these asset classes is given a benchmarket weight to produce the benchmarket portfolio $B$. Say for instance that these weights in equity, fixed income and cash were 60/30/10\% respectively. The portfolio managers portfolio, $P$, on the other hand allocates 70\% to equity, 7\% to fixed income and 27\% to cash. We can examine the difference between these portfolios returns to understand how the asset allocation decision has impacted the performance of $P$ relative to $B$. We use the returns on the benchmark portfolios constituents to identify the outperformance so that the only way we could achieve better is through the weighting. This isolates the contribution from asset allocation.

| Market | Weight in $P$ | Weight in $B$ | Difference | Benchmark Rtn | Contribution |
|:--|:-:|:-:|:-:|:-:|:-:|
| Equity  | 0.7  | 0.6  | 0.1  | 5.81 | 0.5810  |
| Fixed income | 0.07  | 0.3  | -0.23  | 1.45  | -0.3335  |
| Cash | 0.23  | 0.1  |  0.13 |  0.48 | 0.0624  |
| Asset allocation contribution  |   |   |   |   | 0.3099  |

We can see from the example above that the asset allocation decision provides a benefit of around 31 basis points (bps).

A similar exercise can be done with the the individual portfolios we have constructed for each asset class. Say for example that the manager doesn't invest in the S\&P 500 or the Barclay Aggregate index but instead creates their own equity/fixed icome portfolios through a security selection decision. We may can identify the contribution from this decision by comparing the performanc eof the constructed portfolios returns with the benchmarks. For example,

| Market | $P$ Rtn| $B$ Rtn | Difference | Weight | Contribution |
|:--|:-:|:-:|:-:|:-:|:-:|
| Equity  | 7.28  | 5.81  | 1.47  |0.7 | 1.03  |
| Fixed income | 1.89  | 1.45  | 0.44  | 0.07  | 0.03  |
| Asset allocation contribution  |   |   |   |   | 1.06  |

In this case we find that security selection has provided a benfit of 1.06\%. In total, the asset allocation and security selection decisions have provided a benefit of 1.06 + 0.31 = 1.37\% above the benchmark.